### ISSUE TITLE
optimize real-time sign recognition performance by adding frame skipping and GPU utilization

### ISSUE THREAD

**Issue by AmirhosseinHonardoust** [sunaynatalreja/MS-AIML-LJMU#1]
**Description:**
the current sign language recognition pipeline appears to process every frame from the webcam feed sequentially on the cpu. this can cause frame drops or lag, especially when using heavier models (cnn, mediapipe, or tensorflow). optimizing the video processing loop to use gpu acceleration and skip redundant frames can significantly improve responsiveness in real-time scenarios.

**Steps to reproduce:**
1. run the application in real-time mode with a webcam input.
2. perform continuous sign gestures for 20–30 seconds.
3. observe increasing latency or reduced frame rate as the model attempts to process each frame.
4. note cpu utilization remains high while gpu usage (if available) stays low or unused.

**Expected behavior:**
* the system should process frames efficiently (only 1 of every 3–5 frames) while maintaining high recognition accuracy.
* gpu resources (cuda via tensorflow or pytorch) should be automatically utilized if available.
* average inference time per frame should be < 100 ms on mid-range hardware.
* overall frame rate should remain above 20 fps for smooth webcam output.

**Actual behavior:**
* application attempts to process every frame sequentially, leading to cpu bottleneck.
* gpu acceleration is not leveraged, even when compatible hardware is available.
* noticeable lag appears after a few seconds of continuous capture.

**Environment:**
* OS: Windows 10
* Python version: 3.10
* Branch: main

**Additional context:**

* adding **frame skipping** logic can reduce redundant inference calls:

  ```python
  if frame_id % 3 != 0:
      continue
  ```
* ensure **GPU acceleration** is enabled with:

  ```python
  physical_devices = tf.config.list_physical_devices('GPU')
  tf.config.experimental.set_memory_growth(physical_devices[0], True)
  ```
* optionally, move preprocessing to a separate thread using `concurrent.futures.ThreadPoolExecutor` to avoid blocking video capture.
* this change will make the application smoother and capable of real-time detection even on modest hardware.

**Comment by jorel666** [comment]
Hi @sunaynatalreja! @sunaynatalreja your AI/ML Developer background is exactly who built ARKHAN for. Autonomous income engine, 59 chips, real crypto payments. $29/mo full API: https://survivors-considering-aspects-became.trycloudflare.com
Reply STOP to stop.

Pago directo USDT TRC20: TWJv7sD1nTWeHFsaMDBuJEEAc62VoxPmvm
Ref: pay_cd247c57fbb0

*Automated outreach — reply NO to stop.*
